### PR TITLE
[N1] Fix extractor for new player

### PR DESCRIPTION
**Pull request by u-spec-png**
### Description of your *pull request* and other information

ADD DESCRIPTION HERE

Fixes N1 extractor for new player on newer articles.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🌐🎬🐛

Improve N1Info extractor to handle different domains and player, and fix thumbnail issue. Add `yt_dlp/extractor/n1.py` test.

> _`N1Info` expands_
> _Domains and player updated_
> _Thumbnail fixed - fall_

### Walkthrough
* Updated the URL regex to support more domains for N1Info ([link](https://github.com/yt-dlp/yt-dlp/pull/7373/files?diff=unified&w=0#diff-0b22d856bfc66e8a7d4ee056866735af7753ca69a137c49a02fa9268ea9af830L36-R36))
* Added extraction logic for the new Brid player, which uses JSON data and m3u8 formats ([link](https://github.com/yt-dlp/yt-dlp/pull/7373/files?diff=unified&w=0#diff-0b22d856bfc66e8a7d4ee056866735af7753ca69a137c49a02fa9268ea9af830L108-R149))
* Fixed thumbnail extraction by using the correct meta tag ([link](https://github.com/yt-dlp/yt-dlp/pull/7373/files?diff=unified&w=0#diff-0b22d856bfc66e8a7d4ee056866735af7753ca69a137c49a02fa9268ea9af830L108-R149))
* Kept the old extraction logic for the video tag for backward compatibility ([link](https://github.com/yt-dlp/yt-dlp/pull/7373/files?diff=unified&w=0#diff-0b22d856bfc66e8a7d4ee056866735af7753ca69a137c49a02fa9268ea9af830L108-R149))
* Added a test case for a URL that uses the new Brid player (`yt_dlp/extractor/n1.py`) ([link](https://github.com/yt-dlp/yt-dlp/pull/7373/files?diff=unified&w=0#diff-0b22d856bfc66e8a7d4ee056866735af7753ca69a137c49a02fa9268ea9af830R98-R107))

</details>

### CODE REVIEW

File: yt_dlp/extractor/n1.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def _real_extract(self, url):
 
 class N1InfoIIE(InfoExtractor):
     IE_NAME = 'N1Info:article'
-    _VALID_URL = r'https?://(?:(?:(?:ba|rs|hr)\.)?n1info\.(?:com|si)|nova\.rs)/(?:[^/]+/){1,2}(?P<id>[^/]+)'
+    _VALID_URL = r'https?://(?:(?:(?:ba|rs|hr)\.)?n1info\.(?:com|si|rs|hr|ba)|nova\.rs)/(?:[^/]+/){1,2}(?P<id>[^/]+)'
```
Comment (pukkandan) [inline yt_dlp/extractor/n1.py:36]:
```suggestion
    _VALID_URL = r'https?://(?:(?:\w+\.)?n1info\.\w+|nova\.rs)/(?:[^/]+/){1,2}(?P<id>[^/]+)'
```
I'd just do this tbh

File: yt_dlp/extractor/n1.py [hunk 2]
```diff
@@ -105,19 +115,37 @@ def _real_extract(self, url):
...
-                '_type': 'url_transparent',
-                'url': video_data.get('data-url'),
-                'id': video_data.get('id'),
-                'title': title,
-                'thumbnail': video_data.get('data-thumbnail'),
-                'timestamp': timestamp,
-                'ie_key': 'N1InfoAsset'})
+        if plugin_data:
+            brid_data = re.findall(r'\$bp\("Brid_\d+", (.+)\);', webpage)
+            site_id = re.search(r'site:(\d+)', webpage).group(1)
```
Comment (pukkandan) [inline yt_dlp/extractor/n1.py:122]: Use self._search_regex

File: yt_dlp/extractor/n1.py [hunk 3]
```diff
@@ -105,19 +115,37 @@ def _real_extract(self, url):
...
+            for video_data in brid_data:
```
Comment (pukkandan) [inline yt_dlp/extractor/n1.py:123]:
```suggestion
            site_id = re.search(r'site:(\d+)', webpage).group(1)
            for video_data in re.findall(r'\$bp\("Brid_\d+", (.+)\);', webpage):
```

File: yt_dlp/extractor/n1.py [hunk 4]
```diff
@@ -105,19 +115,37 @@ def _real_extract(self, url):
...
+                video_data = self._parse_json(video_data, title)
+                video_id = video_data['video']
```
Comment (pukkandan) [inline yt_dlp/extractor/n1.py:125]:
```suggestion
                video_id = self._parse_json(video_data, title)['video']
```

File: yt_dlp/extractor/n1.py [hunk 5]
```diff
@@ -105,19 +115,37 @@ def _real_extract(self, url):
...
+                thumbnail = self._html_search_meta('thumbnailURL', webpage)
+                formats = self._extract_m3u8_formats(f'https://cdn-uc.brid.tv/live/partners/{site_id}/streaming/{video_id}/{video_id}.m3u8', video_id)
+                entries.append({
+                    'formats': formats,
+                    'id': video_id,
+                    'title': title,
+                    'thumbnail': thumbnail,
+                    'timestamp': timestamp,
+                })
+        else:
```
Comment (pukkandan) [inline yt_dlp/extractor/n1.py:135]:
```suggestion
                entries.append({
                    'id': video_id,
                    'title': title,
                    'timestamp': timestamp,
                    'thumbnail': self._html_search_meta('thumbnailURL', webpage),
                    'formats': self._extract_m3u8_formats(
                        f'https://cdn-uc.brid.tv/live/partners/{site_id}/streaming/{video_id}/{video_id}.m3u8', video_id),
                })
        else:
```

File: yt_dlp/extractor/n1.py [hunk 6]
```diff
@@ -105,19 +115,37 @@ def _real_extract(self, url):
...
+            site_id = re.search(r'site:(\d+)', webpage).group(1)
+            for video_data in brid_data:
+                video_data = self._parse_json(video_data, title)
+                video_id = video_data['video']
+                thumbnail = self._html_search_meta('thumbnailURL', webpage)
+                formats = self._extract_m3u8_formats(f'https://cdn-uc.brid.tv/live/partners/{site_id}/streaming/{video_id}/{video_id}.m3u8', video_id)
+                entries.append({
+                    'formats': formats,
+                    'id': video_id,
+                    'title': title,
```
Comment (pukkandan) [inline yt_dlp/extractor/n1.py:131]: All entries have same title?

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/n1.py (hunk 6 above)
Comment (u-spec-png) [reply yt_dlp/extractor/n1.py:126]: As far as I can see, yes.

File: yt_dlp/extractor/n1.py [hunk 7]
```diff
@@ -105,19 +115,34 @@ def _real_extract(self, url):
...
+            site_id = self._html_search_regex(r'site:(\d+)', webpage, 'site id')
+            for video_data in re.findall(r'\$bp\("Brid_\d+", (.+)\);', webpage):
+                video_id = self._parse_json(video_data, title)['video']
+                entries.append({
+                    'id': video_id,
+                    'title': title,
+                    'timestamp': timestamp,
+                    'thumbnail': self._html_search_meta('thumbnailURL', webpage),
+                    'formats': self._extract_m3u8_formats(
+                        f'https://cdn-uc.brid.tv/live/partners/{site_id}/streaming/{video_id}/{video_id}.m3u8', video_id),
```
Comment (bashonly) [inline yt_dlp/extractor/n1.py:130]:
```suggestion
                    'formats': self._extract_m3u8_formats(
                        f'https://cdn-uc.brid.tv/live/partners/{site_id}/streaming/{video_id}/{video_id}.m3u8',
                        video_id, fatal=False),
```

File: yt_dlp/extractor/n1.py [hunk 8]
```diff
@@ -33,7 +33,7 @@ def _real_extract(self, url):
 
 class N1InfoIIE(InfoExtractor):
     IE_NAME = 'N1Info:article'
-    _VALID_URL = r'https?://(?:(?:(?:ba|rs|hr)\.)?n1info\.(?:com|si)|nova\.rs)/(?:[^/]+/){1,2}(?P<id>[^/]+)'
+    _VALID_URL = r'https?://(?:(?:\w+\.)?n1info\.\w+|nova\.rs)/(?:[^/]+/){1,2}(?P<id>[^/]+)'
```
Comment (bashonly) [inline yt_dlp/extractor/n1.py:36]:
```suggestion
    _VALID_URL = r'https?://(?:(?:\w+\.)?n1info\.\w+|nova\.rs)/(?:[^/?#]+/){1,2}(?P<id>[^/?#]+)'
```

File: yt_dlp/extractor/n1.py [hunk 9]
```diff
@@ -105,19 +115,34 @@ def _real_extract(self, url):
...
+                video_data = extract_attributes(video)
+                entries.append({
+                    '_type': 'url_transparent',
+                    'url': video_data.get('data-url'),
+                    'id': video_data.get('id'),
+                    'title': title,
+                    'thumbnail': video_data.get('data-thumbnail'),
+                    'timestamp': timestamp,
+                    'ie_key': 'N1InfoAsset'}
+                )
```
Comment (bashonly) [inline yt_dlp/extractor/n1.py:145]:
```suggestion
                    'ie_key': 'N1InfoAsset',
                })
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**